blogs/blocks/featured-posts/featured-posts.js

```diff
@@ -14,70 +29,65 @@ export default async function decorate(block) {
 
     const heading = post.querySelector('p > strong');
     const link = post.querySelector('p > a');
+    const entryPlaceholder = {
+      link: link.href,
+      category: heading.textContent,
+    };
 
-    post.innerHTML = `
-        <a href="${link.href}"><div class="picture-placeholder"></div></a>
-        <div>
-          <p class="category">${heading.textContent}</p>
-          <a href="${link.href}"></a>
-        </div>
-    `;
+    post.innerHTML = buildPost(true, entryPlaceholder);
     post.id = `blogs_${heading.textContent.toLowerCase().replace(' ', '_')}`;
 
     // small delay to get to lcp faster
-    setTimeout(() => {
+    setTimeout(async () => {
       if (link.title !== 'auto') {
         getMetadataJson(`${link.href}`).then((meta) => {
-          const image = meta['og:image'];
-          const title = meta['og:title'];
-          if (image) {
-            const pic = createOptimizedPicture(image, '', false, [{ width: '200' }]);
-            post.querySelector('.picture-placeholder').replaceWith(pic);
-          }
-          post.querySelector(':scope > div > a').textContent = title;
+          const imageDefault = meta['og:image'];
+          const titleDefault = meta['og:title'];
+          const entryForDefault = {
+            ...entryPlaceholder,
+            title: titleDefault,
+            pic: imageDefault && createOptimizedPicture(imageDefault, '', false, [{ width: '200' }]),
+          };
+
+          post.innerHTML = buildPost(false, entryForDefault);
           post.classList.remove('post-placeholder');
-          post.querySelector('a > picture').parentElement.setAttribute('title', title);
         });
       } else {
-        fetch('/blogs/query-index.json?limit=500').then((resp) => {
-          if (resp.ok) {
-            resp.json().then((json) => {
-              const topic = getMetadata('topic');
-              const subTopic = getMetadata('subtopic');
-              const mostRecentPost = json.data.find((postData) => {
-                if (!topic) {
-                  // no topic so just get the first one
-                  return true;
-                }
+        const topic = getMetadata('topic');
+        const subTopic = getMetadata('subtopic');
+        const mostRecentPost = await ffetch(indexUrl)
```

Comment:
I think the reason this only works on the home page is due to the param passed here. the url constructed above makes query index always relative to the current page. instead, I think just pass a constant url as a string, eg: ` await ffetch('/blogs/query-index.json')`

Reply:
Fixed